src/app.rs

```diff
@@ -44,6 +50,53 @@ impl EwwWindow {
     pub fn close(self) {
         self.gtk_window.close();
     }
+
+    #[cfg(not(target_os = "macos"))]
+    pub fn set_struts(&self) -> Result<()> {
+        let win: X11Window = self
+            .gtk_window
+            .get_window()
+            .expect("Could not get gdk window from gtk window")
+            .downcast()
+            .expect("Could not get X11 window from gdk window");
+        let xid = win.get_xid() as u32;
+
+        let def_struts = self.definition.struts;
+        let struts: [u32; 12] = [
+            def_struts.left,
+            def_struts.right,
+            def_struts.top,
+            def_struts.bottom,
+            def_struts.left_start_y,
+            def_struts.left_end_y,
+            def_struts.right_start_y,
+            def_struts.right_end_y,
+            def_struts.top_start_x,
+            def_struts.top_end_x,
+            def_struts.bottom_start_x,
+            def_struts.bottom_end_x,
+        ];
+
+        let (conn, _) = RustConnection::connect(None)?;
+        let get_atom = |name: &str| -> Result<_> { Ok(intern_atom(&conn, false, name.as_bytes())?.reply()?.atom) };
```

Comment:
This should just be a free, top-level function. Having it as a closure defined locally makes it seem more magic than it really is.